src/liboslexec/llvm_instance.cpp

```diff
@@ -279,14 +279,7 @@ BackendLLVM::llvm_type_groupdata()
         ++order;
         for (int i = 0; i < nuserdata; ++i) {
             TypeDesc type = types[i];
-            // NB: Userdata derivs are not currently supported in OptiX, since
-            //     making room for them in the GroupData struct can result in a
-            //     large per-thread memory allocation, which could negatively
-            //     impact performance.
-            int n         = (!use_optix()) ? type.numelements()
-                                         * 3  // make room for derivs by default
-                                           : type.numelements();
-            type.arraylen = n;
+            type.arraylen = type.numelements() * 3;  // make room for derivs by default
```

Comment:
Now that I'm looking at this, I think that we should only multiply by 3 if `type.basetype == TypeDesc::FLOAT`. Because anything based on int or string can't possibly have derivatives, so perhaps you can reclaim a little space that way (if indeed anything is ever declared userdata with those types, but it is possible).

Also, I'm seeing a CI complaint about clang-format on this line.

Reply:
Sounds good, I pushed an update addressing both points.